swap_meet/vendor.py

```diff
@@ -1,2 +1,109 @@
 class Vendor:
-    pass
\ No newline at end of file
+    '''
+    Class representing a single vendor with items to trade
+
+    Functions:
+        add(object, item)
+        remove(object, item)
+        get_by_category(object, category)
+        swap_items(object, another object, my_item, their_item)
+        swap_first_item(object, another object)
+        get_best_by_category(self, category):
+        swap_best_by_category(self, other, my_priority, their_priority):
+
+    Variables:
+        inventory: list of items for a vendor to see default is [] 
+    '''
+    def __init__(self, inventory=None):
+        if not inventory == None:
+            self.inventory = inventory
+        else:
+            self.inventory = []
+        # self.inventory = inventory if inventory is not None else []
+
+
+    def add(self, item):
+        """Method to add passed item to instances inventory"""
+        self.inventory.append(item)
+
+        return item
+
+
+    def remove(self, item):
+        """Method to remove passed item to instances inventory"""
+        if not item in self.inventory:
+            return False
+
+        self.inventory.remove(item)
+
+        return item
+
+
+    def get_by_category(self, category):
+        """Method to return a list of items of a passed category"""
+        category_list = []
+
+        for item in self.inventory:
+            if item.category == category:
+                category_list.append(item)
+
+        #category_list = [item for item in self.inventory if item.category == category]
+        return category_list
+
+
+    def swap_items(self, other, my_item, their_item):
+        """Method to pass items of two Vendor Classes"""
+        if (not my_item in self.inventory 
+            or not their_item in other.inventory):
+            return False
+        
+        self.add(their_item)
+        other.add(my_item)
+
+        self.remove(my_item)
+        other.remove(their_item)
+
+        return True
+
+
+    def swap_first_item(self, other):
+        """Method to swap first items of two Vendor Classes"""
+        if self.inventory == [] or other.inventory == []:
+            return False
+
+        self.swap_items(other, self.inventory[0], other.inventory[0])
+
+        return True
```

Comment:
Great use of helper functions! 

Currently, swapping any items has a time complexity of O(n) due to the `remove` method in `swap_items`. We can make a subtle improvement to O(1) by using multiple assignment instead:

```
        self.inventory[0], other.inventory[0] = other.inventory[0], self.inventory[0]
```        

Here's more info on Python multiple assignment: https://www.w3schools.com/python/gloss_python_assign_value_to_multiple_variables.asp